### ISSUE TITLE
Minio: svc type is always LoadBalancer regardless which value is set

### ISSUE THREAD

**Issue by SimonTheLeg** [helm/charts#5669]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Version of Helm and Kubernetes**:
Client: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.8.2", GitCommit:"a80231648a1473929271764b920a8e346f6de844", GitTreeState:"clean"}

1.9.7-gke.0

**Which chart**:
stable/minio

**What happened**:
Trying to install the minio chart always creates a Service of type LoadBalancer (although according to the README and values.yaml the default should be ClusterIP)

example:
```shell
helm install stable/minio --name my-minio --dry-run --debug
```
Will return the following service

```
# Source: minio/templates/minio_svc.yaml
kind: Service
apiVersion: v1
metadata:
  name: my-minio-minio-svc
  labels:
    app: my-minio-minio
    chart: "minio-0.5.7"
    release: "my-minio"
    heritage: "Tiller"
spec:
  type: LoadBalancer
  loadBalancerIP:
  selector:
    app: my-minio-minio
  ports:
    - name: service
      port: 9000
      targetPort: 9000
      protocol: TCP
```
Even trying to override the parameter manually

```shell
helm install stable/minio --set service.type=NodePort --name my-minio --dry-run --debug
```

will still result in a service of type LoadBalancer

```
# Source: minio/templates/minio_svc.yaml
kind: Service
apiVersion: v1
metadata:
  name: my-minio-minio-svc
  labels:
    app: my-minio-minio
    chart: "minio-0.5.7"
    release: "my-minio"
    heritage: "Tiller"
spec:
  type: LoadBalancer
  loadBalancerIP:
  selector:
    app: my-minio-minio
  ports:
    - name: service
      port: 9000
      targetPort: 9000
      protocol: TCP
---
```

**What you expected to happen**:
Without any values the default type should be ClusterIP. When overriding the parameter the type should be the override value.

**How to reproduce it** (as minimally and precisely as possible):
Run
```shell
helm install stable/minio --name my-minio --dry-run --debug
```
Or
```shell
helm install stable/minio --set service.type=NodePort --name my-minio --dry-run --debug
```

**Anything else we need to know**:
What is interesting is, that in the COMPUTED VALUES section of helm debug you can see that it does pick up the service.type but also has another value called serviceType which is always LoadBalancer regardless of what values you set. Overriding 'serviceType' instead of 'service.Type' will result in the creation of the correct svc type.  
```
COMPUTED VALUES:
...
service:
  type: NodePort
servicePort: 9000
serviceType: LoadBalancer
...
```

**Comment by mattjmcnaughton** [comment]
Hi @SimonTheLeg - great question!

I think the issue is that you are using a older version of the `minio` chart. It looks like you are using version `0.5.7`, and the most recent version of the `minio` chart is 1.3.0. The good news is that the bug you mention is fixed in 1.3.0.

I think if you do `helm repo update`, and then try this again, it should work!

**Comment by SimonTheLeg** [comment]
aaaaaaah! Yep now it's working perfectly! :) I was so confused yesterday, but yes that makes sense 🙈😄😄
